src/compas_timber/elements/beam.py

```diff
@@ -406,6 +420,7 @@ def _create_shape(frame, xsize, ysize, zsize):
 
     @reset_computed
```

Comment:
This resets the cached attributes of the class?

Reply:
yes. some `Element` has some internal instance attributes which are cached after the first time they are created. For example `Element._geometry` is one. These need to be cleared when changes are made which might result in new geometry. This decorator takes care of that.